librouteros/query.py

```diff
@@ -71,3 +71,24 @@ def Or(left: QueryGen, right: QueryGen, *rest: QueryGen) -> QueryGen:
     yield from chain.from_iterable(rest)
     yield "?#|"
     yield from ("?#|",) * len(rest)
+
+
+class AsyncQuery:
+    def __init__(self, path, keys: typing.Sequence[Key], api):
+        self.path = path
+        self.keys = keys
+        self.api = api
+        self.query: typing.Tuple[str, ...] = tuple()
+
+    def where(self, *args: str):
+        self.query = tuple(chain.from_iterable(args))
+        return self
+
+    def __aiter__(self):
+        keys = ",".join(str(key) for key in self.keys)
+        keys = f"=.proplist={keys}"
+        cmd = str(self.path.join("print"))
+        return self.api.rawCmd(cmd, keys, *self.query)
+
+    def __iter__(self):
```

Comment:
Since it's async only, no need for `__iter__` method at all.

Reply:
It is prevent anyone from using list expression on it. Just throws an error.

Reply:
If it's not present python should throw `AttributeError` as far as I remember.

Reply:
changed to `AttributeError `